frontend/packages/pipelines-plugin/src/components/pipelines/modals/common/utils.ts

```diff
@@ -84,13 +91,18 @@ export const getPipelineRunData = (
           }),
       annotations,
       namespace: pipeline ? pipeline.metadata.namespace : latestRun.metadata.namespace,
-      labels: _.merge({}, pipeline?.metadata?.labels, latestRun?.metadata?.labels, {
-        'tekton.dev/pipeline': pipelineName,
-      }),
+      labels: _.merge(
+        {},
+        pipeline?.metadata?.labels,
+        latestRun?.metadata?.labels,
+        latestRun.spec.pipelineRef && {
+          'tekton.dev/pipeline': pipelineName,
+        },
+      ),
     },
     spec: {
       ...(latestRun?.spec || {}),
-      pipelineRef: {
+      pipelineRef: latestRun.spec.pipelineRef && {
         name: pipelineName,
       },
```

Comment:
Perhaps avoid adding it if it's not there.
```suggestion
      ...(latestRun.spec.pipelineRef && { pipelineRef: {
        name: pipelineName,
      } }),
```

Reply:
done